kandinsky/src/context_polygon.cpp

```diff
@@ -4,65 +4,66 @@
-        if (pointsY[i] > bottom)
-            bottom = pointsY[i];
+  for (int i = 0; i < numberOfPoints; i++) {
+    if (pointsY[i] < top) {
+      top = pointsY[i];
     }
+    if (pointsY[i] > bottom) {
+      bottom = pointsY[i];
+    }
+  }
 
-    if (top < 0)
-        top = 0;
-    if (bottom > m_clippingRect.height())
-        bottom = m_clippingRect.height();
+  if (top < 0) {
+    top = 0;
+  }
+  if (bottom > m_clippingRect.height()) {
+    bottom = m_clippingRect.height();
+  }
 
 
-    for (int y=top; y<=bottom; y++) {
+  for (int y=top; y<=bottom; y++) {
 
-        int switches=0;
-        KDCoordinate switchesX[KDContext::k_polygonMaxNumberOfPoints];
-        int lastPointIndex = numberOfPoints-1;
+    int switches=0;
+    KDCoordinate switchesX[KDContext::k_polygonMaxNumberOfPoints];
+    int lastPointIndex = numberOfPoints-1;
 
-        for (int i=0; i<numberOfPoints; i++) {
-            if ((pointsY[i]<y && pointsY[lastPointIndex]>=y) || (pointsY[lastPointIndex]< y && pointsY[i]>=y)) 
-                switchesX[switches++] = (int) round(pointsX[i]+1.0*(y-pointsY[i])/(pointsY[lastPointIndex]-pointsY[i])*(pointsX[lastPointIndex]-pointsX[i])); 
-            lastPointIndex=i; 
-        }
+    for (int i=0; i<numberOfPoints; i++) {
+      if ((pointsY[i]<y && pointsY[lastPointIndex]>=y) || (pointsY[lastPointIndex]< y && pointsY[i]>=y)) {
+        switchesX[switches++] = (int) round(pointsX[i]+1.0*(y-pointsY[i])/(pointsY[lastPointIndex]-pointsY[i])*(pointsX[lastPointIndex]-pointsX[i])); 
+      }
+      lastPointIndex=i; 
+    }
 
-        //Sorting switches by a bubble sort
-        int i=0;
-        while (i<switches-1) {
-            if (switchesX[i]>switchesX[i+1]) {
-                KDCoordinate temp = switchesX[i]; 
-                switchesX[i]=switchesX[i+1]; 
-                switchesX[i+1]=temp; 
-                if (i) i--; 
-            }
-            else {
-                i++; 
-            }
-        }
+    //Sorting switches by a bubble sort
+    int i=0;
+    while (i<switches-1) {
+      if (switchesX[i]>switchesX[i+1]) {
+        KDCoordinate temp = switchesX[i]; 
+        switchesX[i]=switchesX[i+1]; 
+        switchesX[i+1]=temp; 
+        if (i) i--; 
+      }
+      else {
+        i++; 
+      }
+    }
 
-        for (i=0; i<switches; i+=2) {
-            if (switchesX[i]>=right) 
-                break;
+    for (i=0; i<switches; i+=2) {
+      if (switchesX[i]>=right) {
+        break;
+      }
 
-            if (switchesX[i+1]>left) {
-                if (switchesX[i]<left) 
-                    switchesX[i]=left;
-                if (switchesX[i+1]>right) 
-                    switchesX[i+1]=right;
-                for (int x=switchesX[i]; x<=switchesX[i+1]; x++) 
-                    setPixel(KDPoint(x,y), color); 
-            }
-        }
+      if (switchesX[i+1]>left) {
+          fillRect( KDRect( switchesX[ i ] , y, switchesX[ i+1 ] - switchesX[ i ], 1 ), color ) ;
+      }
     }
+  }
 
 } 
```

Comment:
```suggestion
} 

```

Reply:
Would you advise me to add this to my last commit and force-push, revert the last commit and redo it with this addition, or make a new commit with just this?

Reply:
Tou should probably commit that, and then squash it with the previous commit